addons/website_slides/tests/test_slide_channel.py

```diff
@@ -274,6 +275,47 @@ def _get_completion_time_pdf(*args, **kwargs):
         self.assertEqual(13.37, round(slides_1.completion_time, 2))
         self.assertEqual(123.0, slides_2.completion_time)
 
+    def test_if_unlink_and_unpublish_channel_slides_sends_competition_email(self):
+        """When all content (slide.slide) are deleted from the course (slide.channel),
+        the course participants (slide.channel.partner) SHALL NOT receive completers email notification. """
+
+        with self.mock_mail_gateway():
+            _count_send_mails = lambda: len(self._new_mails.filtered(lambda m: m.model == 'slide.channel.partner'))
+
+            # 0. Setup: make sure there's channel with slides, that nobody completed
+            self.assertTrue(len(self.channel.slide_ids) > 1, "Channel shall have at least two slides.")
+            self.assertTrue(len(self.channel.channel_partner_ids) > 0, "Channel shall have at least one participant (channel_partner_ids)")
+            self.assertTrue(_count_send_mails() == 0, "Someone received competition mail notification (nobody should have)")
+
+            # 1.1. Action: Unpublish one slide
+            slide_published = self.channel.slide_ids.search([('channel_id', '=', self.channel.id), ('is_published', '=', True)], limit=1)
```

Comment:
Aren't all slides of a channel with said `channel_id` by definition?
Also, do you *need* to use `search` (db query, while the record may have been in the cache)?
Why not simply
```suggestion
            slide_published = self.channel.slide_ids.filtered('published')[:1]
```

Reply:
good catch, thanks!